core/src/main/java/com/graphhopper/routing/profiles/TagParserFactory.java

```diff
@@ -0,0 +1,495 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.profiles;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.util.AbstractFlagEncoder;
+import com.graphhopper.routing.util.spatialrules.SpatialRule;
+import com.graphhopper.routing.util.spatialrules.SpatialRuleLookup;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.shapes.GHPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class TagParserFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TagParserFactory.class);
+    public static final ReaderWayFilter ACCEPT_IF_HIGHWAY = new ReaderWayFilter() {
+        @Override
+        public boolean accept(ReaderWay way) {
+            return way.getTag("highway") != null;
+        }
+    };
+
+    private static final Set<String> fwdOneways = new HashSet<String>() {
+        {
+            add("yes");
+            add("true");
+            add("1");
+        }
+    };
+
+    public static final String ROUNDABOUT = "roundabout";
+    public static final String ROAD_CLASS = "road_class";
+    public static final String ROAD_ENVIRONMENT = "road_environment";
```

Comment:
IMHO the word `road_environment` is not ideal here. From the name I would assume that it's about the environment of the road, like city, forest, desert, etc.

Naming it is hard though. Maybe `road_structure` or `road_attributes`?

Reply:
Especially this name is hard, yes. But attributes is not what is meant ... and structure also not my preference ;) ... let us think again :)